### PR TITLE
[SIMPLE_FORMS] feat: updates to submission builder in accordance with new direction

### PR DESCRIPTION
**Pull request by pennja**
## Summary

- This work updates the submission archive builder in accordance with VBA direction

## Related issue(s)

- [Create Solution for adding Form Submission Remediation Documents to an S3 bucket](https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1633)

## Testing done

- [x] *New code is covered by unit tests*

## Requested Feedback

Any

### CODE REVIEW

File: modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb [hunk 1]
```diff
@@ -129,8 +116,12 @@ def form_data_hash
         @form_data_hash ||= JSON.parse(submission.form_data)
       end
 
+      # Name the form PDFs and/or individual submission folders
+      # uniquely, using a field that also appears in the manifest.
+      # The recommended format is Form-number-vagov-submission ID
       def submission_pdf_filename
-        @submission_pdf_filename ||= "form_#{form_data_hash['form_number']}.pdf"
+        form_number = form_data_hash['form_number']
+        @submission_pdf_filename ||= "form_#{form_number}_vagov_#{benefits_intake_uuid}.pdf"
```
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb:124]: I'm using underscores here given that both the form_number and benefits_intake_uuid will contain dashes.

File: modules/simple_forms_api/app/services/simple_forms_api/s3/utils.rb [hunk 2]
```diff
@@ -29,7 +29,7 @@ def handle_error(message, error, **)
       end
 
       def temp_directory_path
-        @temp_directory_path ||= Rails.root.join("tmp/#{benefits_intake_uuid}-#{SecureRandom.hex}/").to_s
+        @temp_directory_path ||= Rails.root.join("tmp/#{benefits_intake_uuid}-#{SecureRandom.hex}-archive/").to_s
```
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/s3/utils.rb:32]: Ensuring there are no collisions with similar temp folder/file logic elsewhere.

**Review by Thrillberg — APPROVED** [review]
Makes sense to me! Glad to see the options get simplified. 😄

**Comment by ryan-mcneil** [comment]
@pennja it's probably worth merging master after the changes we made

**Review by ryan-mcneil — APPROVED** [review]

**Pull request merged**